orangecontrib/text/widgets/owdocumentembedding.py

```diff
@@ -0,0 +1,236 @@
+
+        self.language_cb = comboBox(widget = widget_box,
+                                    master = self,
+                                    value = 'language',
+                                    label = 'Language: ',
+                                    orientation = Qt.Horizontal,
+                                    items = self.languages,
+                                    callback = self._option_changed
+        )
+
+        self.aggregator_cb = comboBox(widget = widget_box,
+                                      master = self,
+                                      value = 'aggregator',
+                                      label = 'Aggregator: ',
+                                      orientation = Qt.Horizontal,
+                                      items = self.aggregators,
+                                      callback = self._option_changed 
+        )
+
+        self.auto_commit_widget = auto_commit(widget = self.controlArea,
+                                              master = self,
+                                              value = '_auto_apply',
+                                              label = 'Apply',
+                                              commit = self.commit)
+
+        self.cancel_button = QPushButton(
+                                    'Cancel',
+                                    icon = self.style()
+                                    .standardIcon(QStyle.SP_DialogCancelButton))
+
+        self.cancel_button.clicked.connect(self.cancel)
+
+        hbox = hBox(self.controlArea)
+        hbox.layout().addWidget(self.cancel_button)
+        self.cancel_button.setDisabled(True)
+
+    def set_input_corpus_summary(self, corpus):
+        if(corpus is None):
+            self.info.set_input_summary(self.info.NoInput)
+        else:
+            self.info.set_input_summary("{} documents."
+                                        .format(len(corpus)))
+
+    def set_output_corpus_summary(self, corpus):
+        if(corpus is None):
+            self.info.set_output_summary(self.info.NoOutput)
+        else:
+            unsuccessful = np.sum(np.isnan(corpus.X)) / EMB_DIM
+            if(unsuccessful > 0):
+                self.Warning.unsuccessful_embeddings()
+            self.info.set_output_summary("Successful: {}, Unsuccessful: {}"
+                                        .format(len(corpus) - unsuccessful,
+                                                 unsuccessful))
+
+    @Inputs.corpus
+    def set_data(self, data):
+        self.Warning.clear()
+        self.set_input_corpus_summary(data)
+        
+        if (not data):
+            self.corpus = None
+            self.clear_outputs()
+            return
+
+        self.corpus = data
+        self.commit()
+
+    def _option_changed(self):
+        self.commit()
+
+    def commit(self):
+        if(self.corpus is None):
+            self.clear_outputs()
+            return
+
+        self._set_fields(False)
+
+        try:
```

Comment:
start never raise the error so it should not be caught here all the errors are raised through on_exception, move the check for connection error there.